[Bug] PulsarStandalone started with error if `--stream-storage-port` is not 4181

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.


### Version

- master
- 3.3.0

### Minimal reproduce step

Run `PULSAR_STANDALONE_USE_ZOOKEEPER=1 bin/pulsar standalone --stream-storage-port 4182`


### What did you expect to see?

PulsarStandalone started without error.

### What did you see instead?

Following error appears in console repeatedly.
```
2024-07-02T22:36:31,943+0800 [client-scheduler-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.clients.impl.internal.LocationClientImpl - Not able to locate storage container
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:539) ~[io.grpc-grpc-api-1.56.0.jar:1.56.0]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:548) ~[io.grpc-grpc-stub-1.56.0.jar:1.56.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:567) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:71) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:735) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:716) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[io.grpc-grpc-core-1.56.0.jar:1.56.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: io.grpc.netty.shaded.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:4181
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[?:?]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946) ~[?:?]
	at io.grpc.netty.shaded.io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	at io.grpc.netty.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[io.grpc-grpc-netty-shaded-1.56.0.jar:1.56.0]
	... 1 more
```

Main thread keeps waiting
```
"main" #1 prio=5 os_prio=31 cpu=1278.18ms elapsed=520.07s tid=0x000000012a016400 nid=0x2203 waiting on condition  [0x000000016c072000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.11/Native Method)
        - parking to wait for  <0x000020000bf03bd0> (a java.util.concurrent.CompletableFuture$Signaller)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.11/LockSupport.java:252)
        at java.util.concurrent.CompletableFuture$Signaller.block(java.base@17.0.11/CompletableFuture.java:1866)
        at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@17.0.11/ForkJoinPool.java:3465)
        at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.11/ForkJoinPool.java:3436)
        at java.util.concurrent.CompletableFuture.timedGet(java.base@17.0.11/CompletableFuture.java:1939)
        at java.util.concurrent.CompletableFuture.get(java.base@17.0.11/CompletableFuture.java:2095)
        at org.apache.bookkeeper.common.concurrent.FutureUtils.result(FutureUtils.java:80)
        at org.apache.bookkeeper.common.concurrent.FutureUtils.result(FutureUtils.java:61)
        at org.apache.pulsar.zookeeper.LocalBookkeeperEnsemble.runStreamStorage(LocalBookkeeperEnsemble.java:372)
        at org.apache.pulsar.zookeeper.LocalBookkeeperEnsemble.startStandalone(LocalBookkeeperEnsemble.java:437)
        at org.apache.pulsar.PulsarStandalone.startBookieWithZookeeper(PulsarStandalone.java:474)
        at org.apache.pulsar.PulsarStandalone.start(PulsarStandalone.java:302)
        at org.apache.pulsar.PulsarStandaloneStarter.main(PulsarStandaloneStarter.java:149)
```

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/pulsar-broker/src/main/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsemble.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsemble.java
@@ -360,7 +360,7 @@ public class LocalBookkeeperEnsemble {
         // create a default namespace
         try (StorageAdminClient admin = StorageClientBuilder.newBuilder()
              .withSettings(StorageClientSettings.newBuilder()
-                 .serviceUri("bk://localhost:4181")
+                 .serviceUri("bk://localhost:" + streamStoragePort)
                  .backoffPolicy(Backoff.Jitter.of(
                      Type.EXPONENTIAL,
                      1000,

--- a/pulsar-broker/src/test/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsembleTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsembleTest.java
@@ -21,6 +21,8 @@ package org.apache.pulsar.zookeeper;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
+
+import org.apache.bookkeeper.conf.ServerConfiguration;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
@@ -53,5 +55,19 @@ public class LocalBookkeeperEnsembleTest {
         assertFalse(ensemble.getZkServer().isRunning());
         assertFalse(ensemble.getZkClient().getState().isConnected());
         assertFalse(ensemble.getBookies()[0].isRunning());
+    }
+
+    @Test(timeOut = 10_000)
+    public void testStartWithSpecifiedStreamStoragePort() throws Exception {
+        LocalBookkeeperEnsemble ensemble = null;
+        try {
+            ensemble =
+                    new LocalBookkeeperEnsemble(1, 0, 0, 4182, null, null, true, null);
+            ensemble.startStandalone(new ServerConfiguration(), true);
+        } finally {
+            if (ensemble != null) {
+                ensemble.stop();
+            }
+        }
     }
 }

